components/Dialogs/tests/unit/MDCAlertControllerAlertThemerTests.swift

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+import MaterialComponents.MDCAlertScheme
+import MaterialComponents.MDCAlertThemer
+
+class MDCAlertControllerAlertThemerTests: XCTestCase {
+
```

Comment:
_Style Nit:_ 


```
// Variable on the class
var alertScheme: MDCAlertScheme!

func setup() {
  super.setup()

  alertScheme = MDCAlertScheme()
}
```

Then remove all `let alertScheme = MDCAlertScheme()`

Reply:
A forced unwrapped mutable instance variable seemed like a worse solution than repeating one line in each test which can then be local and immutable. I typically use setup for larger/more-complex initialization logic.

Reply:
+1 to Cody's suggestion. Boilerplate set-up should be in `-setUp`, especially if it's used by all of the tests.  I would suggest the same for anything found in "most" tests if it doesn't cause problems with the tests that don't require it.

For example, initializing a ColorScheme variable even if not all tests need to use `colorScheme`.

Reply:
This line is not boilerplate. It's what being tested, so taking it out will reduce the readability of the tests.  In one case it takes out the one line that is the "When" part of the test.  Are you sure you want to change that?

Reply:
As far as I can see, all of the `let alertScheme  = MDCAlertScheme()` is contained within a `// Given` block.

Reply:
So final decision was to keep that line inside the function for the single case when it's what's being tested. For the other cases, I pulled the line out into the setup function. Thanks for everyone's input on this.